src/tests/ftest/container/boundary.py

```diff
@@ -0,0 +1,144 @@
+            rank = self.params.get("rank", '/run/container/execute_io/*')
+            obj_classs = self.params.get("obj_classs", '/run/container/execute_io/*')
+            try:
+                data_bytes = container.execute_io(io_run_time, rank, obj_classs)
+                self.log.info(
+                    "===(3.%d.%d)Wrote %d bytes to container %s", pool_num, cont_num,
+                    data_bytes, container)
+            except (DaosTestError, TestFail) as err:
+                msg = "#(3.{}.{}) container IO failed, err: {}".format(pool_num, cont_num, err)
+                self.fail(msg)
+        time.sleep(2)  #to syncup containers before close
+
+        try:
+            self.log.info("===(4.%d.%d)create_container_and_test, container closing.",
+                pool_num, cont_num)
+            container.close()
+            self.log.info("===(4.%d.%d)create_container_and_test, container closed.",
+                pool_num, cont_num)
+        except (DaosTestError, TestFail) as err:
+            msg = "#(4.{}.{}) container close fail, err: {}".format(pool_num, cont_num, err)
+            self.fail(msg)
+
+    def create_containers(self, pool=None, pool_num=10, num_containers=100, with_io=False):
+        """To create number of containers parallelly on pool.
+        Args:
+            pool(str): pool handle.
+            pool_num (int): pool number to create containers.
+            num_containers (int): number of containers to create.
+            with_io (bool): enable container test with execute_io.
+        """
+
+        self.log.info("==(2.%d)create_containers start.", pool_num)
+        thread_manager = ThreadManager(self.create_container_and_test, self.timeout - 30)
+
+        for cont_num in range(num_containers):
+            thread_manager.add(
+                pool=pool, pool_num=pool_num, cont_num=cont_num, with_io=with_io)
+        self.log.info("==(2.%d) %d containers created.", pool_num, num_containers)
+
+        # Launch the create_container_and_test threads
+        self.log.info("==Launching %d create_container_and_test threads", thread_manager.qty)
+        failed_thread_count = thread_manager.check_run()
+        if failed_thread_count > 0:
+            msg = "#(2.{}) FAILED create_container_and_test Threads".format(failed_thread_count)
+            self.d_log.error(msg)
+            self.fail(msg)
+
+    def create_pools(self, num_pools=10, num_containers=100, with_io=False):
+        """To create number of pools and containers parallelly.
+        Args:
+            num_pools (int): number of pools to create.
+            num_containers (int): number of containers to create.
+            with_io (bool): enable container test with execute_io.
+        """
+
+        # Setup the thread manager
+        thread_manager = ThreadManager(self.create_containers, self.timeout - 30)
+
+        for pool_number in range(num_pools):
+            pool = self.get_pool()
```

Comment:
Step 1 says "Create pools by ThreadManager", but you're creating it serially here, not with a thread.

Reply:
Thanks for the comment, I updated the steps.

Reply:
I think all `ThreadManager` references, etc. can still be removed. Since `self.get_pool` creates the pool

Reply:
This is correct, create pools, and create containers with io test by Threadmanager.

Reply:
Oops. I misread that!